### DISCUSSION TITLE
Maybe move `ssl` imports into a `TYPE_CHECKING` block?

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [encode/httpx#3263] [General]
I made [a package to patch httpx for using in pyodide](https://github.com/pyodide/pyodide/discussions/4999), and you can try [a live demo here](https://py3.online/console#ZnJvbSBtaWNyb3BpcCBpbXBvcnQgaW5zdGFsbAphd2FpdCBpbnN0YWxsKFsicHlvZGlkZS1odHRweCIsICJzc2wiXSkKCmZyb20gcHlvZGlkZV9odHRweCBpbXBvcnQgKgpwYXRjaF9odHRweCgpCgpmcm9tIGh0dHB4IGltcG9ydCBnZXQKZ2V0KCJodHRwczovL2h0dHBiaW4ub3JnL2pzb24iKQoKcHJpbnQoXy50ZXh0KQpfLmpzb24oKQo=).

The problem is, code inside `httpx` have imports to `ssl`, so I have to install `ssl` (in pyodide runtime, these modules not bundled by default). But `ssl` is heavy and [I only see type imports inside `httpx` with `ssl`](https://github.com/search?q=repo%3Aencode%2Fhttpx%20ssl&type=code), so I wonder can try this:

```py
if TYPE_CHECKING:
    import ssl

# original: def load_ssl_context_no_verify(self) -> ssl.SSLContext:
def load_ssl_context_no_verify(self) -> "ssl.SSLContext":
```

Or

```py
from __future__ import annotations

if TYPE_CHECKING:
    import ssl

def load_ssl_context_no_verify(self) -> ssl.SSLContext:
```

**Comment by gyanu2507** [comment]
Current httpx already does the TYPE_CHECKING split in `httpx/_config.py`. The module-level `import ssl` is behind `TYPE_CHECKING`, and `create_ssl_context()` does `import ssl` only when you actually build a context.

So HTTPS still needs `ssl` at runtime. Importing httpx for a type annotation should not.

If Pyodide still pulls `ssl` on `import httpx`, check **httpcore** (and any `httpx[http2]` extra). Those can import OpenSSL bits at import time even when httpx itself is lazy.

You cannot drop the runtime import entirely. `ssl.SSLContext` is used for real connections, not just types.
